<?xml version="1.0"?>
<launch>
    <arg name="sensor_model" description="RobosenseHelios/RobosenseHelios16P/RobosenseBpearl/RobosenseM1Plus"/>
    <arg name="return_mode" default="Dual" description="See readme for supported return modes"/>
    <arg name="frame_id"  default="robosense"/>

    <arg name="calibration_file" default="$(find-pkg-share nebula_decoders)/calibration/robosense/$(var sensor_model).csv"/>
    <arg name="correction_file" default="$(find-pkg-share nebula_decoders)/calibration/robosense/$(var sensor_model).dat"/>

    <arg name="sensor_ip" default="192.168.1.201" description="Lidar Sensor IP"/>
    <arg name="host_ip" default="255.255.255.255" description="Broadcast IP from Sensor"/>
    <arg name="data_port" default="6699" description="LiDAR Data Port"/>
    <arg name="difop_port" default="7788" description="LiDAR Difop Port"/>
    <arg name="packet_mtu_size" default="1500" description="Packet MTU size"/>
    <arg name="rotation_speed" default="600" description="Motor RPM, the sensor's internal spin rate."/>
    <arg name="cloud_min_angle" default="0" description="Field of View, start degrees."/>
    <arg name="cloud_max_angle" default="360" description="Field of View, end degrees."/>
    <arg name="setup_sensor" default="True" description="Enable sensor setup on hw-driver."/>

    <node pkg="nebula_ros" exec="robosense_driver_ros_wrapper_node"
          name="robosense_cloud" output="screen">
        <param name="sensor_model" value="$(var sensor_model)"/>
        <param name="return_mode" value="$(var return_mode)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="calibration_file" value="$(var calibration_file)"/>
        <param name="correction_file" value="$(var correction_file)"/>
    </node>

    <node pkg="nebula_ros" exec="robosense_hw_interface_ros_wrapper_node"
          name="robosense_hw_driver" output="screen">
        <param name="sensor_model" value="$(var sensor_model)"/>
        <param name="return_mode" value="$(var return_mode)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="sensor_ip" value="$(var sensor_ip)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="host_ip" value="$(var host_ip)"/>
        <param name="data_port" value="$(var data_port)"/>
        <param name="difop_port" value="$(var difop_port)"/>
        <param name="packet_mtu_size" value="$(var packet_mtu_size)"/>
        <param name="rotation_speed" value="$(var rotation_speed)"/>
        <param name="cloud_min_angle" value="$(var cloud_min_angle)"/>
        <param name="cloud_max_angle" value="$(var cloud_max_angle)"/>
        <param name="setup_sensor" value="$(var setup_sensor)"/>
    </node>

    <node pkg="nebula_ros" exec="robosense_hw_monitor_ros_wrapper_node"
          name="robosense_hw_monitor_$(var frame_id)" output="screen">
        <param name="sensor_model" value="$(var sensor_model)"/>
        <param name="return_mode" value="$(var return_mode)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="sensor_ip" value="$(var sensor_ip)"/>
        <param name="frame_id" value="$(var frame_id)"/>
        <param name="host_ip" value="$(var host_ip)"/>
        <param name="data_port" value="$(var data_port)"/>
        <param name="difop_port" value="$(var difop_port)"/>
        <param name="packet_mtu_size" value="$(var packet_mtu_size)"/>
        <param name="rotation_speed" value="$(var rotation_speed)"/>
        <param name="cloud_min_angle" value="$(var cloud_min_angle)"/>
        <param name="cloud_max_angle" value="$(var cloud_max_angle)"/>
    </node>

</launch>
